src/custom/state/profile/hooks.tsx

```diff
@@ -0,0 +1,28 @@
+import { useCallback } from 'react'
+import { closeAnnouncement } from './actions'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+
+export function useAnnouncementVisible(contentHash?: string): boolean {
+  const announcementVisible = useAppSelector((state) => state.profile.announcementVisible)
+
+  // No hash, no visible
+  if (!contentHash) {
+    return false
+  }
+
+  // If the hash has been closed, will return false,
+  // if its a new hash returns true (visible)
+  return announcementVisible[contentHash] ?? true
```

Comment:
does `return !!announcementVisible[contentHash]` also not work?

Reply:
but why is that better? 
same thing but more inefficient


well, actually is not the same thing, here the default is `true`, the equivalent code to what u mentioned would be `announcementVisible[contentHash] ?? false`